test/metabase/server/middleware/auth_test.clj

```diff
@@ -38,11 +38,13 @@
 (deftest wrap-current-user-info-test
   (testing "Valid requests should add `metabase-user-id` to requests with valid session info"
     (let [session-id (random-session-id)]
-      (tt/with-temp Session [_ {:id      session-id
-                                :user_id (test.users/user->id :rasta)}]
+      (try
+        (db/insert! Session {:id      session-id
+                             :user_id (test.users/user->id :rasta)})
         (is (= (test.users/user->id :rasta)
                (-> (auth-enforced-handler (request-with-session-id session-id))
-                   :metabase-user-id))))))
+                   :metabase-user-id)))
+        (finally (db/delete! Session :id session-id)))))
```

Comment:
Since the User isn't deleted after this test I had to manually delete the Session in a `finally` form... not ideal but before this PR it actually wasn't getting deleted at all for H2